### PR TITLE
Enhancement: token option rendering

### LINKED ISSUE #58: Delete Token ATM Configuration Cause Unexpected Dashboard Interactions
After deleting Token ATM configuration using dashboard, other dashboard functionalities behaved in an unexpected way since they assume the existence of a valid Token ATM configuration. A possible solution to this issue is to navigate the user to course selection page after the deletion of Token ATM configuration.

Originally discovered by @epsilonError in [#57 (comments)](https://github.com/UCI-IN4MATX-191-Token-ATM/token-atm-spa/pull/57#issuecomment-1635239638).

### PR DESCRIPTION
**Pull request by yongkanm**
### Description

1.  The “description” column in the table that describes token options for token option groups now has both a minimum width and a maximum width to ensure an acceptable degree of readability. If the horizontal space is insufficient, the table will become horizontally scrollable. 
2.  The table that describes token options for each token option group are now moved from the description of the first quiz question to the assignment description. It will be appended after the configured description of the token option group. If a quiz for a token option group is created in the old version, saving the token option group metadata (name and description) or any token option in it will apply this change.
3.  A page named “Token ATM - Available Token Options” that shows all published token option groups and their available token options for students.
    1.  If the page does not exist yet, any operation that causes a token option group being created/deleted/updated/published/unpublished will trigger the creation of the page.
    2.  If a published token option group does not contain any available token options, a message indicating this fact will be shown for that group in this page.
4.  After the user deletes Token ATM configuration for a course using the “Delete All” button, the user will be navigated to the course selection page.

### Testing Note

Please test if enhancements mentioned above work as expected.

### CODE REVIEW

File: src/app/services/token-atm-configuration-manager.service.ts [hunk 1]
```diff
@@ -263,6 +312,7 @@ export class TokenATMConfigurationManagerService {
             TokenATMConfigurationManagerService.TOKEN_ATM_SECURE_CONFIGURATION_PAGE_NAME,
             `<p>${JSON.stringify(configuration.getSecureConfig())}</p>`
         );
+        this.updateAvailableTokenOptionPage(configuration);
```
[the remark this answers is not in the document]
Comment (yongkanm) [reply src/app/services/token-atm-configuration-manager.service.ts:315]: Yes, there should be an `await`. A fix will be included in the next commit. Thanks for catching that!

**Comment by yongkanm** [comment]
> **Note**: The default string/directions in Question 1 for Token ATM Request Submission Surveys appears even if no options are available to students to choose from.
> 
> When a Token Option Group is created and has no token options (or all token options in that group haven't been migrated), the table of options and actual multiple choice selections do not exist. But the Q1 directions `Make a request by choosing an option below (see the table above for the detailed description of each option)` do appear.

Thanks for catching that. The question description is modified to indicate the fact that no request can be made when there is no available token options in a token option group.

**Comment by yongkanm** [comment]
> **Note**: The display multiple token option groups on the "Available Token Options" page are placed very near to each other.
> 
> The h3 tag with the group name can appear closer, or at least as close, to the table above as to the table below.
> 
> Adding spacing between the Token Option Groups would be an improvement. (A quick change would be adding a `<p>`, `<hr>`, or `<br>` tag could be a quick fix. And if the spacing is added to the top of the div, then the spacing could also work for the same table in the survey quiz.)

Paddings are added to both the top of each token option group name in the "Available Token Options" page and the top of the information table for token option in the assignment description.

**Comment by yongkanm** [comment]
> **Note**: The long-strings of text in the description cell in the token option group table are wrapped differently on Canvas Pages and Canvas Quizzes.
> 
> Long unbroken strings of text in Canvas Quizzes can extend out to the right of its description cell. The same strings and cells on Canvas Pages are wrapped and the height of the table row is increased.

This is an unavoidable Canvas behavior. Canvas applies `word-wrap: break-word` on page content but not on quiz. Also, Canvas has [a whitelist of inline CSS style for HTML content](https://community.canvaslms.com/t5/Canvas-Resource-Documents/Canvas-HTML-Editor-Allowlist/ta-p/387066). Since no style in the whitelist could control the word breaking behavior, there is no CSS fix for this issue.

This issue could still be fixed by calculating the word break in Token ATM and forcefully apply it using elements like `<br>`. However, this workaround cannot make accommodation for user-side settings, and it could also cause negative impact on the accessibility. In my opinion, since this issue will only appear when there is a non-breakable word longer than approximately 50 characters (the max width limitation of description column), we don't need to fix that.

**Comment by yongkanm** [comment]
> **Note**: If an error occurs during the the deletion of Token ATM content, once the error modal is closed the redirection to the Course Selection page still happens.
> 
> Since the course context of the page may change after this error, we may need to add some heuristics about whether or not to still auto-redirect. Or at least mention the specific course or other identifying info to the error message.

Thanks for catching this usability issue. Adding heuristics to it could be troublesome since the Token ATM configuration is most likely broken for this course if an error occurs. I've added the course name to the error message and indicate that the user will be redirected once they close this modal.

**Review by epsilonError — DISMISSED** [review]

File: src/app/services/token-atm-configuration-manager.service.ts [hunk 2]
```diff
@@ -156,11 +157,15 @@ export class TokenATMConfigurationManagerService {
             courseId,
             quizId,
             TokenATMConfigurationManagerService.TOKEN_ATM_QUIZ_PREFIX + tokenOptionGroup.name,
-            String.raw`<div>${tokenOptionGroup.description}</div><div>${dataTable}</div>`
+            String.raw`<div>${tokenOptionGroup.description}</div><div style="padding-top: 1em;">${dataTable}</div>`
         );
+        const questionPrompt =
+            tokenOptionGroup.availableTokenOptions.length == 0
+                ? "There isn't anything you can make a request to via this quiz. If you think this is an error, please contact your instructor."
```
[the remark this answers is not in the document]
Comment (yongkanm) [reply src/app/services/token-atm-configuration-manager.service.ts]: Change is made in the new commit.

**Review by epsilonError — APPROVED** [review]

**Pull request merged**